mbs_results/constrains.py

```diff
@@ -0,0 +1,170 @@
+import operator
+from typing import List
+
+import pandas as pd
+
+
+def replace_values_index_based(
+    df: pd.DataFrame, target: str, a: int, compare: str, b: int
+) -> None:
+    """
+    Perform comparisons between a subset of df with a and subset of df with b and
+    replace target from a with target from b when comparison is met. Both a
+    and b must exist in the first level index, the comparison is based on the
+    remaining indices.
+
+    Note that this function does not return anything, it modifies the input
+    dataframe.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Original dataframe to replace the values.
+    target : str
+        Column name for values to be replced.
+    a : int
+        Question_no to check.
+    compare : str
+        Logical operator to compare, accepted '>', '<','>=','<=','=='.
+    b : int
+        Question_no to check against.
+    """
+    # For improved perfomance
+    df.sort_index(inplace=True)
```

Comment:
Is this going to cause a bigger issue when we have lots more data to process? Whats the slowdown like if unsorted? (Just questions to think about not needing to change)

Reply:
It was a pandas warning suggesting this , afterwards we do an index comparison to get matches for question no, i didn't notice any difference , perhaps some comparison is needed